pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -117,21 +117,6 @@ func (c *Controller) Remove(b *v3.EtcdBackup) (runtime.Object, error) {
 	if err := c.etcdRemoveSnapshotWithBackoff(b); err != nil {
 		logrus.Warnf("giving up on deleting backup [%s]: %v", b.Name, err)
 	}
-	if b.Spec.BackupConfig.S3BackupConfig == nil {
-		return b, nil
-	}
-	// try to remove from s3 for 3 times, then give up. if we don't we get stuck forever
-	var delErr error
```

Comment:
@moelsayed can you confirm that this logic used to be exercised only for recurring snapshot use case